### PR TITLE
Converting demos into tests 

### LINKED ISSUE #10: cmake build for demos
I get the following error when I try to build and run demos :
I use the command cmake --build for building source files there , is there any additional setting I should do before build?
"REAL_BBTEST_SOURCES

/usr/bin/ld: cannot find -lpipert
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/simple.dir/build.make:84: simple] Error 1
make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/simple.dir/all] Error 2
make: *** [Makefile:84: all] Error 2 "

### PR DESCRIPTION
**Pull request by martinb214**
Fixes #10

### CODE REVIEW

File: test/SchedulerTest.cpp [hunk 1]
```diff
@@ -74,3 +107,41 @@ TYPED_TEST(SchedulerTest, ScheduledChannelCreationWithTypes) {
...
+                                        std::bind(&HumanPrinter::PrintName, &hp, std::placeholders::_1));
+
+  sch.Start();
+
+  pipert::Timer::Time time = pipert::Timer::time();
+  for(int i = 0; i < 10; i++) {
+    pipert::PacketToFill<Human> packet_to_fill = sc1.Acquire("Reverser", time, "Jimi Hendrix", 28 + i);
+  }
+
+  usleep(1000);
```
Comment (BenBlaise) [inline test/SchedulerTest.cpp:135]: The "how-long-should-we-sleep" question also emerged in PR #27, just a heads up.

**Review by BenBlaise — COMMENTED** [review]
Everything looks fine to me. Thanks for clearing up the demo's that were not working. It will save newcomers the mental overhead of figuring out that `main.cpp` and `join.cpp` were most probably pseudocode / not currently operable, and only `main2.cpp` showed proper usage.

File: test/SchedulerTest.cpp (hunk 1 above)
Comment (martinb214) [reply test/SchedulerTest.cpp:135]: I saw that, thanks. I'm following that topic to get the proper way and replace this solution.

Comment (gerazo) [reply test/SchedulerTest.cpp:135]: See my reply there. I believe that a yield in a loop is the best solution for a test.

Comment (martinb214) [reply test/SchedulerTest.cpp]: Added new solution in a fixup.

**Review by Hossam-Abdin — APPROVED** [review]
The whole testing is very good and very understandable, I like also to mention what great documentation we have now which helped me to understand the scheduler more.

Things we can test:
- stopping the scheduler 2 times in a row.

Question:
- is yielding the threads before stopping is necessary because the documentation says that the scheduler can stop it from my understanding

**Comment by BenBlaise** [comment]
Replying to @Hossam-Abdin:

The way I understand it, when we are yielding towards the end of `SchedulerPipeLineTest`, we are doing so because we cannot be sure when exactly would all the packets sent into the pipeline arrive at the `PolledChannel` we are taking them off of. Instead of waiting for a fixed duration, we do a soft version of busy-waiting by letting go of the rest of the current timeslice allotted to us by the OS scheduler if all packets have not arrived already, to check again in the next timeslice. This avoids wasting performance at the expense of a slight delay.

On testing the scheduler being stopped twice in a row:

See `StartStopRestartTest`. When I was writing this one, I had the same idea too, but it seems that currently when we call to stop the scheduler, we have an assert that fails if it was not running before. So, in the debug mode tests, this would just fail everytime.

**Review by BenBlaise — APPROVED** [review]
Everything looks like it's ready to be merged. It really seems like yielding in a loop is an elegant solution to the undetermined runtime of the test.

**Comment by gerazo** [comment]
@BenBlaise @martinb214 Is this ready to go? If yes, let's do a final merge from master and go!

**Comment by martinb214** [comment]
@Hossam-Abdin:
@BenBlaise is right about yielding. Waiting a little is not a bad practice at all, but it's a bit random and can increase the delay.
  
@gerazo:
It's ready and final merge is done. This error in CI about python3-lxml package is very strange. Maybe we should rerun these checks.

**Pull request merged**